### PR TITLE
node: implement more validators in native code

### PR DESCRIPTION
**Pull request by nektro**
reduces the overhead of these functions and the amount of duplicate js implementations across the builtin modules

pulled out of https://github.com/oven-sh/bun/pull/13757 to merge independently

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 305fad9f86d372422e7db5531e2ae87a3e570030 is building: [`#3961`](https://buildkite.com/bun/bun/builds/3961)

File: src/js/builtins/ProcessObjectInternals.ts [hunk 1]
```diff
@@ -244,13 +244,7 @@ export function initializeNextTickQueue(process, nextTickQueue, drainMicrotasksF
   var drainMicrotasks = drainMicrotasksFn;
   var reportUncaughtException = reportUncaughtExceptionFn;
 
-  function validateFunction(cb) {
-    if (typeof cb !== "function") {
-      const err = new TypeError(`The "callback" argument must be of type "function". Received type ${typeof cb}`);
-      err.code = "ERR_INVALID_ARG_TYPE";
-      throw err;
-    }
-  }
```
Comment (nektro) [inline src/js/builtins/ProcessObjectInternals.ts:253]: the regular one also takes a `name` parameter. in this one it's hardcoded to be `"callback"` so the callsite needs to be updated

File: src/bun.js/bindings/NodeValidator.cpp [hunk 2]
```diff
@@ -163,4 +171,268 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateFunction, (JSC::JSGlobalObject * glo
...
+    return JSValue::encode(jsUndefined());
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsFunction_validatePlainFunction, (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto value = callFrame->argument(0);
+    auto name = callFrame->argument(0);
```
Comment (dylan-conway) [inline src/bun.js/bindings/NodeValidator.cpp]:
```suggestion
    auto name = callFrame->argument(1);
```

Comment (dylan-conway) [reply src/bun.js/bindings/NodeValidator.cpp]: oh, is this intentional?

File: src/bun.js/bindings/NodeValidator.cpp [hunk 3]
```diff
@@ -163,4 +171,268 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateFunction, (JSC::JSGlobalObject * glo
...
+    return JSValue::encode(jsUndefined());
+}
+
+JSC_DEFINE_HOST_FUNCTION(jsFunction_validateBuffer, (JSC::JSGlobalObject * globalObject, JSC::CallFrame* callFrame))
+{
+    JSC::VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto buffer = callFrame->argument(0);
+    auto name = callFrame->argument(0);
```
Comment (dylan-conway) [inline src/bun.js/bindings/NodeValidator.cpp]: same here

File: src/bun.js/bindings/NodeValidator.cpp [hunk 4]
```diff
@@ -163,4 +171,268 @@ JSC_DEFINE_HOST_FUNCTION(jsFunction_validateFunction, (JSC::JSGlobalObject * glo
...
```
Comment (dylan-conway) [inline src/bun.js/bindings/NodeValidator.cpp]:
```suggestion
    auto name = callFrame->argument(1);
```

**Review by dylan-conway — APPROVED** [review]

**Pull request merged**